accepted/targeting-packs-and-runtime-packs.md

```diff
@@ -0,0 +1,287 @@
+* Globally resolveable without project context (depended upon by certain VS features)
+
+**Traditional targeting pack weaknesses**
+* Windows-only acquisition
+* Admin-only acquisition
+* Manual installation
+* Infamous error experience when not installed
+* No self-contained deployment strategy
+
+**Framework NuGet package stregnths**
+* Cross-platform/cross-IDE acquisition
+* Non-admin acquisition
+* Seamless acquisition: just clone, restore, and build
+   * Users can target a minimum patch version of a .NET Core shared framework
+
+
+**Framework Nuget package weaknesses**
+* Slow build (restore) performance
+* Extremely complex NuGet graph that leaks to customers
+* Large size on disk
+* Difficult to understand distinction between platform and ecosystem
+* Difficult to predict when network access is required
+* Requires project context for resolution
+
+
+# Summary of New Approach
+
+We introduce a third, hybrid approach to framework references that aims to capture the strengths of the earlier approaches while mitigating weaknesses. 
+
+In this new approach, framework reference assemblies will be commonly resolved from a global location alongside `dotnet`. This is similar in spirit to the classic targeting packs in `C:\Program Files\Reference Assemblies`, but cross-platform and compatible with non-admin SDK deployment.
+
+On top of that, we layer a fallback to acquiring the same assets via NuGet. This will be used so that you can easily build projects targeting frameworks that are older than the SDKs you have installed. 
+
+Finally, we add a new `FrameworkReference` concept to the build, to represent the use of a group of assemblies that version with the target framework.
+
+
+# Targeting, Runtime, and AppHost Packs
+
+We use the term "pack" for a collection of files used by the build. A pack can either be deployed globally alongside `dotnet` or wrapped in a NuGet package.
+
+There are three categories of packs:
+
+1. Targeting packs - reference assemblies, documentation, and other design-time assets
+2. Runtime packs - runtime assets for self-contained publish
+3. AppHost packs - app executable template to generate native app executable
+
+
+# Globally Installed Packs 
+
+Targeting packs (and AppHost packs) will be bundled with the SDK, and available to be installed globally via native installers. This bundling is directly analagous to how the shared frameworks are bundled with the SDK or installable individually.
+
+A new `packs/` folder will be added next to `dotnet` alongside `shared/` and `sdk/`. Only the smaller targeting packs and apphost pack will be bundled here, and the larger runtime packs will always be acquired via NuGet.
+
+```
+dotnet[.exe]
+    shared/    <-- shared framework implementation for execution
+        Microsoft.NETCore.App/3.0.0/...
+        Microsoft.ASPNetCore.App/3.0.0/...
+
+    packs/     <-- shared framework design-time assets for compilation
+        Microsoft.NETCore.App.Ref/3.0.0/...
```

Comment:
... here will change the same as the nuget package layout. This makes the code between finding in this folder and transporting from nuget have fewer special cases. So with the current "data/" option still being discussed, this would include data/ here.

Reply:
See https://github.com/dotnet/designs/pull/50#discussion_r258569041

This will include data/(txt|xml files for sdk) + remaining files in nuget conventions ref/netcoreapp3.0/(reference assemblies), runtimes/(rid)/lib/(tfm)/(impl assemblies), runtimes/(rid)/native/(native binaries).

It is an error for (tfm) or (rid) to not match the corresponding part of the package name.